src/v2i-hub/PedestrianPlugin/src/PedestrianPlugin.cpp

```diff
@@ -143,54 +168,60 @@ void PedestrianPlugin::UpdateConfigSettings()
 	std::lock_guard<mutex> lock(_cfgLock);
-
-	GetConfigValue<string>("WebServiceIP",webip);
-	GetConfigValue<uint16_t>("WebServicePort",webport);
-	GetConfigValue<string>("WebSocketHost",webSocketIP);
-	GetConfigValue<string>("WebSocketPort",webSocketURLExt);
+	GetConfigValue<std::string>("WebServiceIP", webip);
+	GetConfigValue<uint16_t>("WebServicePort", webport);
+	GetConfigValue<std::string>("WebSocketHost", webSocketIP);
+	GetConfigValue<std::string>("WebSocketPort", webSocketURLExt);
 	GetConfigValue<int>("Instance", instance);
-	GetConfigValue<string>("DataProvider", dataprovider);
-	GetConfigValue<float>("FLIRCameraRotation",cameraRotation);
-	GetConfigValue<string>("HostString",hostString);
-
-	PLOG(logDEBUG) << "Pedestrian data provider: "<< dataprovider.c_str() << std::endl;
-	PLOG(logDEBUG) << "Before creating websocket to: " << webSocketIP.c_str() <<  " on port: " << webSocketURLExt.c_str() << std::endl;
+	GetConfigValue<std::string>("DataProvider", dataprovider);
+	GetConfigValue<float>("FLIRCameraRotation", cameraRotation);
+	GetConfigValue<std::string>("HostString", hostString);
 
+	PLOG(logDEBUG) << "Pedestrian data provider: " << dataprovider;
+	
 	if (dataprovider.compare("FLIR") == 0)
-	{
-		try
-		{
-			std::thread webthread(&PedestrianPlugin::StartWebSocket,this);
-			PLOG(logDEBUG) << "Thread started!!: " << std::endl;
-			
-			webthread.detach(); // wait for the thread to finish
-
-			std::thread xmlThread(&PedestrianPlugin::checkXML,this);
-			PLOG(logDEBUG) << "XML Thread started!!: " << std::endl;
-			
-			xmlThread.detach(); // wait for the thread to finish
+    {
+		PLOG(logDEBUG) << "Before creating websocket to: " << webSocketIP.c_str() <<  " on port: " << webSocketURLExt.c_str();
+		
+        StopWebService();
+        if (!runningWebSocket)
+        {
+			PLOG(logDEBUG) << "Starting WebSocket Thread";
+            std::thread webSocketThread(&PedestrianPlugin::StartWebSocket, this);
+            PLOG(logDEBUG) << "WebSocket Thread started!!";
+
+			PLOG(logDEBUG) << "Starting XML Thread";
+			std::thread xmlThread(&PedestrianPlugin::checkXML, this);
+			PLOG(logDEBUG) << "XML Thread started!!";
+			webSocketThread.join(); // wait for the thread to finish
+			xmlThread.join(); // wait for the thread to finish
+        }
+    }
 
-		}
-		catch(const std::exception& e)
-		{
-			PLOG(logERROR) << "Error connecting to websocket: " << e.what() << std::endl;
-		}
-				
-			
-	}
-	else  // default if PSM XML data consumed using the webservice implementation
+	else if (dataprovider.compare("PSM") == 0) // default if PSM XML data consumed using the webservice implementation
+	{
+        StopWebSocket();
+        if (!runningWebService)
+        {
+			PLOG(logDEBUG) << "Starting WebService Thread";
+            std::thread webServiceThread(&PedestrianPlugin::StartWebService, this);
+			PLOG(logDEBUG) << "WebService Thread started";
+			webServiceThread.join(); // wait for the thread to finish
+        }
+    }
+	else
```

Comment:
Can we have some sort of WARNING log to indicate that the dataprovider configuration is misconfigured. Otherwise someone could have a typo and have no indication why the PSM plugin is not working as intended. Ex
```
PLOG (logWARNING) << "Invalid configured data provider. Valid values include FLIR and PSM!";
```

Reply:
Updated